central/role/datastore/datastore_impl.go

```diff
@@ -136,6 +140,14 @@ func (ds *dataStoreImpl) RemoveRole(ctx context.Context, name string) error {
 	return ds.roleStorage.Delete(ctx, name)
 }
 
+func verifyRoleOriginMatches(ctx context.Context, role *storage.Role) error {
+	if !declarativeconfig.IsOriginModifiable(ctx, role.GetTraits().GetOrigin()) {
+		return errors.Wrapf(errox.InvalidArgs, "role %q's origin is %s, cannot be modified or deleted within this context",
```

Comment:
Same notion within the other PR:
I would be in favor of `errox.NotAuthorized` + improved error message (i.e. dropping "within this context" and adding information regarding the permission).